tests/test_inference_async_client.py

```diff
@@ -0,0 +1,138 @@
+"""Contains tests for AsyncInferenceClient.
```

Comment:
This should have the copyright as well

Reply:
Should we test that both the async and sync clients output the exact same things, but with one being synchronous and the other being asynchronous? (maybe that's already the case).

If you intend on the two clients having the same methods, with the same signatures, would it make sense to include a test for that?

Reply:
> Should we test that both the async and sync clients output the exact same things, but with one being synchronous and the other being asynchronous?

Yes that's the case for `text_generation` and `sentence_similarity` tests. I chose to test those two since text_generation is by far the "most complicated" code. I don't think it's worth testing every single task as long as we know that they are generated via the exact same code.

About exact values, that's kinda what we are doing. But since the tests are using VCR (i.e. not calling the production endpoint), it's not really a complete test anyway. I tried making VCR load the same yaml file for both a sync and a async test but I didn't manage to make it work with VCR.py. In the end I don't think it's worth investigating to much in this. 

> If you intend on the two clients having the same methods, with the same signatures, would it make sense to include a test for that?

I can add such a test!